### DISCUSSION TITLE
Model for task with missing data and response deadline?

### DISCUSSION THREAD

**Discussion by chutcher** [lnccbrown/HSSM#593] [Q&A]
Hi,

I have a task that I am working to model with a DDM, but for which decisions had to be made within 3 seconds, leading to a fair number of missing responses. I've been looking at the HSSM documentation and see that in a previous release (v 0.2.1) the ability to model missing responses and deadlines was added. However, I am not 100% clear on how to actually take advantage of this functionality. Based on this discussion (https://github.com/lnccbrown/HSSM/issues/173) I gather that I might need to add a deadline column to my data frame, presumably specifying a 3 second limit, and perhaps use one of the onnx files, along with a missing_data=True flag in my model and the path to the onnx file provided as a likelihood function. But I am not sure which onnx file might be most appropriate, and am a little uncertain whether my other assumptions are correct. Apologies if I've missed the documentation or further discussion already posted somewhere, but I'd love some advice!

Thanks in advance,
Cendri

**Comment by AlexanderFengler** [comment]
Hi @chutcher, 

We front-loaded a bit of the HSSM infrastructure on this, but let it lay dormant while focusing on some other issues in the meanwhile, e.g. the documentation part.

The idea is that you can provide data with missing (coded) values and use a `[model]_deadline` model string and the backend will then adjust the likelihood computations accordingly. 

What happens internally is that the missing values are going to be assigned to a network that computes the probability of omission given the parameters and deadline for a given trial, so this will in principle work for any of the models.

We can prioritize adding more docs on this functionality inc. a tutorial within a week or two if that helps.


Best,
Alex

**Reply by chutcher** [reply]
Hi Alex,

Hope all's well! I'm finally getting back to this question and would love to re-up this discussion. I'm very keen to get a version of code working that can handle missed responses due to deadlines. I'm currently doing the following, but it's not working, so I'm assuming I've misunderstood how to specify this properly:

  model_task = hssm.HSSM(
      data=pp_data,
      model="ddm",
      z=0.5,
      include=[
          {
              "name": "v",
              "prior": {
                  "Intercept": {
                      "name": "Normal",
                      "mu": 0.0,
                      "sigma": 0.5,
                  },
              },
              "formula": "v ~ 1 + (1|participant_id)",
              "link": "identity",
          },
          {
              "name": "a",
              "prior": {
                  "Intercept": {
                      "name": "HalfNormal",
                      "sigma": 1.5,
                  },
              },
              "formula": "a ~ 1 + (1|participant_id)",
              "link": "identity",
          }
      ],
      deadline=True,
      loglik_missing_data="/Users/me/TimePressure/Analysis/ddm_opn.onnx",
  )

  jax.config.update("jax_enable_x64", False)
  task_infer = model_task.sample(
      sampler="nuts_numpyro", chains=4, cores=2, draws=4000, tune=4000
  )

For what it's worth, my participant dataframe does have a column labeled 'deadline', but that doesn't seem to be the issue.

Thanks in advance!
Cendri

**Reply by AlexanderFengler** [reply]
Hey @chutcher , 

sorry for the late response here, I had missed the message (please don't hesitate to ping via email if you don't get a response here within a few days). 

I will look at this case and report back.

**Reply by chutcher** [reply]
That would be very much appreciated! Thanks so much, Alex!
